cloud/amazon/ec2.py

```diff
@@ -1353,10 +1353,13 @@ def main():
     if not HAS_BOTO:
         module.fail_json(msg='boto required for this module')
 
-    ec2 = ec2_connect(module)
-
     region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module)
 
+    try:
+        ec2 = connect_to_aws(boto.ec2, region, **aws_connect_params)
```

Comment:
Note that the previous line uses **aws_connect_kwargs

You're effectively passing it a dict from somewhere else